source/php/Component/Iframe/Iframe.php

```diff
@@ -98,26 +98,45 @@ private function setSupplierDataAttributes(string $src, array $data)
 
         return $this->data;
     }
-    public function buildEmbedUrl($src)
+    private function buildEmbedUrl($src)
     {
         $srcParsed = parse_url($src);
 
-        $scheme = $srcParsed['scheme'] ?? 'https';
-
         switch ($srcParsed['host']) {
             case 'youtube.com':
             case 'www.youtube.com':
-                $srcParsed['path'] = '/embed/'; // Replace any existing path with /embed/
-
-                parse_str($srcParsed['query'], $query);
-                if (isset($query['v'])) {
-                    $srcParsed['path'] .= $query['v'];
+                /* Replacing the path with /embed/ and then adding the v query parameter to the path and removing the v parameter from the query string. */
+                $srcParsed['host'] = 'youtube.com';
+                $srcParsed['path'] = '/embed/';
+
+                if (isset($srcParsed['query'])) {
+                    parse_str($srcParsed['query'], $query);
+                    if (isset($query['v'])) {
```

Comment:
@sebastianthulin Should perhaps the component actually strip any query vars set in the url? So that settings such as autoplay, looping, sizes a.s.o are set via module settings if they are to be able to be set?

Reply:
See resolve here: https://github.com/helsingborg-stad/component-library/pull/124/commits/e1092ae5078754bf317c67642f77acc14cb8f0f9